packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/cluster_client.ts

```diff
@@ -146,4 +161,22 @@ export class ClusterClient implements ICustomClusterClient {
       ...scopedHeaders,
     };
   }
+
+  private getSecondaryAuthHeaders(request: ScopeableRequest): Headers {
+    const headerSource = isRealRequest(request)
+      ? this.authHeaders?.get(request) ?? {}
+      : request.headers;
+    const authorizationHeader = Object.entries(headerSource).find(([key, _]) => {
+      return key.toLowerCase() === AUTHORIZATION_HEADER;
+    });
+    const scopedHeaders = authorizationHeader
+      ? { [ES_SECONDARY_AUTH_HEADER]: authorizationHeader[1] }
+      : {};
```

Comment:
__question:__ I'm wondering if we should throw (or at least log) an error here to ensure we're not silently ignoring invalid usage?

Reply:
As discussed in slack: we're not doing it for the logic returning the primary auth headers, so I'm scared of changing the behavior there, but we can absolutely do it for the new API. Will add.

Reply:
Thanks!